Apps/GatewayApi/src/Services/GatewayApiMappingService.cs

```diff
@@ -141,6 +142,25 @@ public UserPreferenceModel MapToUserPreferenceModel(UserPreference source)
             return mapper.Map<UserPreference, UserPreferenceModel>(source);
         }
 
+        /// <inheritdoc/>
+        public UserProfile MapToUserProfile(string hdid, Guid termsOfServiceId, DateTime lastLoginDateTime, string? email = null, int? yearOfBirth = null)
+        {
+            return new()
+            {
+                HdId = hdid,
+                IdentityManagementId = null,
+                TermsOfServiceId = termsOfServiceId,
+                Email = email,
+                SmsNumber = null,
+                CreatedBy = hdid,
+                UpdatedBy = hdid,
+                LastLoginDateTime = lastLoginDateTime,
+                EncryptionKey = cryptoDelegate.GenerateKey(),
+                YearOfBirth = yearOfBirth,
+                LastLoginClientCode = authenticationDelegate.FetchAuthenticatedUserClientType(),
+            };
+        }
```

Comment:
I moved it here to avoid DI Constructor parameter issue.  If we move it out and back to where it was, then we will have to add pragma ignore warning on the  constructor.  We are at the max right now with RegistrationService and UserProfileServiceV2 and UserProfileModelService.

Reply:
I think this is an instance where it's better to go over the limit in UserProfileModelService.  I couldn't see us splitting up its functionality further or joining any of its dependencies together.

Reply:
[Moved initialize user profile back to RegistrationService.  Added pragma to ignore constructor warnings.](https://github.com/bcgov/healthgateway/pull/6293/commits/23e9a78d9fe2c48d1a89d6a0dde770170b6d5121)